spyder/plugins/explorer/tests/test_explorer.py

```diff
@@ -46,5 +65,146 @@ def test_project_explorer(project_explorer):
     assert project_explorer
 
 
+def create_folder_files(file_paths, project_dir):
+    """Function to create folders and files to be used in test functions."""
+    list_paths = []
+    for item in file_paths:
+        #  ignore empty string
+        if item:
+            #  create folders
+            if osp.splitext(item)[1]:
+                if osp.split(item)[0]:
+                    dirs, fname = osp.split(item)
+                    os.makedirs(dirs, exist_ok=True)
```

Comment:
`exist_ok` is not available in Python 2.7. It was introduced in 3.2.
https://docs.python.org/3.7/library/os.html#os.makedirs

Reply:
Thank you very much, I removed it, lets see if it passes in your servers, was passing locally

Reply:
It would be simpler to just protect it under a `if not osp.exists(dir)`: condition.

Reply:
Besides, in that case, you want the test to fail there in the code if the operation doesn't succeed, so you don't want to pass any `Exception.

Reply:
`if not osp.exists()` cannot be used, the directories should be created recursively `(maindir/subdir1/subdir2/...)`

Reply:
Okay, good point. I didn't consider it critical since the test will fail anyway pretty quickly if the files and dirs aren't created properly, while the ``makedirs`` function is minor test setup code, not what's being tested, but its better to fail immediately ofc.

Reply:
I thought that was a problem, but I don't think it actually is since ``makdirs()`` only fails if the *last* dir on the path exists, at least in my testing.

Reply:
Yes, indeed it fails only if the last directory exists. The full path should not be similar to any existing full path.

Reply:
@jnsebgosselin could you mark this as resolved! I followed your suggestion!

> It would be simpler to just protect it under a if not osp.exists(dir): condition.